### DISCUSSION TITLE
Storytel reader app

### DISCUSSION THREAD

**Discussion by dnoesgaard** [ntherning/StorytelReaderMods#2] [Q&A] [answered]
I was wondering if the Storytel reader app would still be functional allowing me to access and read books from Storytel after modding? Also what would be the steps needed to revert this mod if needed?

**Comment by ntherning** [comment, marked as answer]
Yes, you can still use the Storytel Reader app after the modding. It will show up as an app in RelaunchX (provided that you went through the Storytel device setup wizard before modding it). See this screenshot: https://github.com/ntherning/StorytelReaderMods/blob/main/images/gen2-running-relaunchx.jpg. Also, at boot the device will ask whether you want to run the original launcher (called `bootstrap` or similar) or RelaunchX. If you choose `bootstrap` the device will go into the Storytel Reader App until you reboot it.

To revert the mod you would need to write back the original `boot.img` file which you should have backed up before flashing the mod as per the instructions in the [Quickstart](https://github.com/ntherning/StorytelReaderMods/tree/main#quickstart). Any app you installed via F-Droid or ADB will still be on the data partition. You can perform a factory reset (`wipe data/factory reset` option) via the [Recovery mode](https://github.com/ntherning/StorytelReaderMods/tree/main#recovery-mode). So, by flashing the backed up `boot.img` and then performing a factory reset from within Recovery Mode your device should be back to a mostly pristine and unmodded state. There might still be some traces of the modding though in some partition which I am not aware of.

To make it possible to revert the device back to exactly the state it had before the mod, you can back up [all partitions](https://github.com/ntherning/StorytelReaderMods/tree/main#partitions), not just `boot`:

```
rkflashtool r boot > /some/safe/place/boot.img
rkflashtool r misc > /some/safe/place/misc.img
rkflashtool r kernel > /some/safe/place/kernel.img
...
```

To revert the mod:

```
rkflashtool w boot < /some/safe/place/boot.img
rkflashtool w misc < /some/safe/place/misc.img
rkflashtool r kernel > /some/safe/place/kernel.img
...
```

If you apply the mod please report back and share your experience here in Discussions. Would be great to know which OS your are on and what apps you managed to run on your device. Thanks!

**Comment by dnoesgaard** [comment]
Thanks! I managed to apply the mod by following your instructions. I struggled to get the device into maskrom mode, and I don't know if the reason was the timing or my cable. Anyway, it worked after a few tries. Being a complete Android noob, I then had to download the SDK Platform-Tools to get ADB. Now I've got F-Droid running, downloaded RelaunchX, but something still doesn seem quite right. The Applications button in RelaunchX doesn't work and if I press the physical orange button on the device, the Storytel app takes over again. I wonder if I messed up somehow? 

Btw, I've got a 2nd gen device and I'm running MacOS.

**Reply by dnoesgaard** [reply]
Actually, turns out I just needed to restart the device to boot up RelaunchX properly. All good!

**Reply by ntherning** [reply]
Nice, I am happy to hear it worked out for you! I will add a note to the guide that a reboot might be required after installing RelaunchX. Thanks!

**Reply by dnoesgaard** [reply]
Sounds good! A link (e.g. https://developer.android.com/tools/releases/platform-tools) to download adb would also have saved me time :)
